cli/src/command/generate_template.rs

```diff
@@ -34,17 +38,25 @@ pub struct GenerateTemplate {
     /// Number of validators in the network.
     #[structopt(long)]
     pub validators_count: u32,
+    /// Supervisor service mode. Possible options are "simple" and "decentralized".
+    ///
+    /// Default: simple.
+    #[structopt(long)]
+    pub supervisor_mode: Option<SupervisorMode>,
```

Comment:
Is `Option` really needed here ?

Reply:
It is an optional parameter. We can make it required though, I do not have any arguments for either way

Reply:
I think making it required makes more sense. Overall, we can't assume what kind of supervisor user wants, except for the `run-dev` mode.